### PR TITLE
Connection Rework Phase Two

### PR DESCRIPTION
**Pull request by Windchild292**
This is the second phase of the connection rework, and is largely a reorganization PR. It is following up from #3596, and also split off of #3450. Specifically, this PR removes the outdated PacketTool, applies the reorganization changes from #3450, moves the PacketReadState Enum to its own file, and updates the ConnectionListener. It also fixes some bugs, both pre-existing and because of #3596, and carries over most of the unrelated cleanup changes from #3450.

### CODE REVIEW

**Comment by Windchild292** [comment]
This is ready for review.

File: megamek/src/megamek/client/bot/princess/Precognition.java [hunk 1]
```diff
@@ -277,7 +281,8 @@ private void pause() {
         while (!getWaiting().get() && !getDone().get()) {
             try {
                 Thread.sleep(100);
-            } catch (InterruptedException ignored) {
+            } catch (Exception ignored) {
+
```
Comment (rjhancock) [inline megamek/src/megamek/client/bot/princess/Precognition.java:285]: Just catching for the loop?

File: megamek/src/megamek/server/Server.java [hunk 2]
```diff
@@ -12807,9 +12805,10 @@ else if ((waterDepth > 0)
                     synchronized (cfrPacketQueue) {
                         try {
                             cfrPacketQueue.wait();
-                        } catch (InterruptedException ignored) {
-                            // Do nothing
+                        } catch (Exception ignored) {
+
```
Comment (rjhancock) [inline megamek/src/megamek/server/Server.java:12809]: Empty catch?

File: megamek/src/megamek/server/Server.java [hunk 3]
```diff
@@ -13888,16 +13888,15 @@ private void manuallyAssignAMSTarget(Entity e,
...
             } else {
                 // Send a client feedback request
                 sendAMSAssignCFR(e, ams, vAttacksInArc);
                 synchronized (cfrPacketQueue) {
                     try {
                         cfrPacketQueue.wait();
-                    } catch (InterruptedException ignored) {
-                        // Do nothing
+                    } catch (Exception ignored) {
+
```
Comment (rjhancock) [inline megamek/src/megamek/server/Server.java:13899]: Empty Catch?

**Review by rjhancock — APPROVED** [review]
Overall looks good.

File: megamek/src/megamek/client/bot/princess/Precognition.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/client/bot/princess/Precognition.java:285]: It's a thread delay, so just ignoring the exception is fine here. I will be looking into any ignored exceptions once the full logging migration is done to see if we are ignoring things that actually make sense to log instead.

Comment (rjhancock) [reply megamek/src/megamek/client/bot/princess/Precognition.java:285]: Ahh. Thought as much.

File: megamek/src/megamek/server/Server.java (hunk 2 above)
Comment (Windchild292) [reply megamek/src/megamek/server/Server.java:12809]: Another thread wait, fine to swallow the exception here

File: megamek/src/megamek/server/Server.java (hunk 3 above)
Comment (Windchild292) [reply megamek/src/megamek/server/Server.java:13899]: Same

**Pull request merged**